packages/wrangler/src/utils/onKeyPress.ts

```diff
@@ -0,0 +1,43 @@
+import readline from "readline";
+import { Transform } from "stream";
+
+export type KeypressEvent = {
+	name: string;
+	sequence: string;
+	ctrl: boolean;
+	meta: boolean;
+	shift: boolean;
+};
+
+export function onKeyPress(callback: (key: KeypressEvent) => void) {
+	// Listening for events on process.stdin (eg .on('keypress')) causes it to go into 'old mode'
+	// which keeps this nodejs process alive even after calling .off('keypress')
+	// WORKAROUND: piping stdin via a transform stream allows us to call stream.destroy()
+	// which then allows this nodejs process to close cleanly
+	// https://nodejs.org/api/process.html#signal-events:~:text=be%20used%20in-,%22old%22%20mode,-that%20is%20compatible
+	const stream = new Transform({
+		transform(chunk, encoding, cb) {
+			cb(null, chunk);
+		},
+	});
+	process.stdin.pipe(stream);
+
+	if (process.stdin.isTTY) {
+		readline.emitKeypressEvents(stream);
+		process.stdin.setRawMode(true);
```

Comment:
Should we capture what the current raw mode is to reuse in the teardown function?

Reply:
This reminds me I need to test this with the auth/account prompts

Reply:
Looks like we'll need to setRawMode(true) again after prompts have been used

(my manual testing just now found that prompts work fine but the hotkeys don't work after)

Reply:
https://github.com/cloudflare/workers-sdk/pull/6149/commits/ebafd22d74ed6f73a8011e16e8d1414dd31b6cde